pkg/process/checks/discovery_check.go

```diff
@@ -0,0 +1,112 @@
+package checks
```

Comment:
nit, mind renaming the file to `process_discovery_check.go` and the test one as well 🙇‍♂️